src/test/java/com/avides/spring/rabbit/listener/OptionalResponseSpringRabbitListenerTest.java

```diff
@@ -0,0 +1,161 @@
+package com.avides.spring.rabbit.listener;
+
+import static org.easymock.EasyMock.eq;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.mock;
+import static org.powermock.api.easymock.PowerMock.replayAll;
+import static org.powermock.api.easymock.PowerMock.verifyAll;
+
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.easymock.annotation.MockStrict;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import org.springframework.amqp.core.MessagePostProcessor;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("javax.management.*")
+public class OptionalResponseSpringRabbitListenerTest
+{
+    private OptionalResponseSpringRabbitListener<Object> successRabbitListener;
+
+    private OptionalResponseSpringRabbitListener<Object> failureRabbitListener;
+
+    @MockStrict
+    private RabbitTemplate responseRabbitTemplate;
+
+    @MockStrict
+    private MeterRegistry meterRegistry;
+
+    @Before
+    public void setup()
+    {
+        successRabbitListener = new SuccessOptionalSpringRabbitListener();
+        Whitebox.setInternalState(successRabbitListener, meterRegistry);
+
+        failureRabbitListener = new FailureOptionalSpringRabbitListener();
+        Whitebox.setInternalState(failureRabbitListener, meterRegistry);
+    }
+
+    @Test
+    public void testHandleEventWithoutReplyTo()
+    {
+        Tags tags = Tags.of(Tag.of("listener", "SuccessOptionalSpringRabbitListener"), Tag.of("from", "sender-app"));
```

Comment:
var, betrifft auch noch weitere Stellen in dieser Klasse.

Reply:
Angepasst.